src/hotspot/share/memory/heapShared.cpp

```diff
@@ -525,10 +650,91 @@ void HeapShared::serialize_subgraph_info_table_header(SerializeClosure* soc) {
   _run_time_subgraph_info_table.serialize_header(soc);
 }
 
+static void verify_the_heap(Klass* k, const char* which) {
+  if (VerifyArchivedFields) {
+    ResourceMark rm;
+    log_info(cds, heap)("Verify heap %s initializing static field(s) in %s",
+                        which, k->external_name());
+    VM_Verify verify_op;
+    VMThread::execute(&verify_op);
+    if (!FLAG_IS_DEFAULT(VerifyArchivedFields)) {
```

Comment:
Comment says "command line", so this should be FLAG_IS_CMDLINE rather than !FLAG_IS_DEFAULT.